pmpro-approvals.php

```diff
@@ -1588,6 +1589,62 @@ public static function pmpro_email_filter( $email ) {
 
 	}
 
+	/**
+	 * Include user fields in the admin approval emails automatically.
+	 *
+	 * @since TBD
+	 * 
+	 * @param object $email The email object.
+	 */
+	public static function pmpro_include_user_fields_in_approvals_emails( $email ) {
+
+		// Bail if we don't have an $email object or the template is missing from the object.
+		if ( empty( $email ) || ! is_object( $email ) || ! isset( $email->template ) ) {
+			return;
+		}
+
+		// If the email is an admin email, include user fields for all Approval emails.
+		if ( ! in_array( $email->template, array( 'admin_approved', 'admin_denied', 'admin_notification_approval' ) ) ) {
+			return;
+		}
+
+		// Get the user_id from the email
+		$user = get_user_by( 'email', $email->data['user_email'] );
+		$user_id = $user ? $user->ID : 0;
+
+		if ( ! empty( $user_id ) ) {
+			//add to bottom of email
+			$field_groups = PMPro_Field_Group::get_all();
+			if ( ! empty( $field_groups ) ) {
+				$fields_content = "<p>" . esc_html__( 'Extra Fields:', 'paid-memberships-pro' ) . "<br />";
+				$added_field = false;
+				// Loop through all the field groups.
+				foreach( $field_groups as $group_name => $group ) {
+					// Loop through all the fields in the group.
+					$fields = $group->get_fields_to_display(
+						array(
+							'scope' => 'checkout',
+							'user_id' => $user_id,
+						)
+					);
+					foreach( $fields as $field ) {
+						$fields_content .= "- " . esc_html( $field->label ) . ": ";
+						$fields_content .= $field->displayValue( get_user_meta( $user_id, $field->name, true), false );
+						$fields_content .= "<br />";
+						$added_field = true;
+					}
```

Comment:
This can all be simplified to using `$field_group->display()`.

We have a commit with similar simplification here: https://github.com/strangerstudios/pmpro-signup-shortcode/commit/341dbd1c19f01329ee4ca1648951bddb55abfbc5

Reply:
The `$field_group->display()` tries to render the HTML fields for processing and this does not work when adding to the email's content.

A possible solution is to adjust the 'scope' to support 'values-only' or similar that bypasses all logic to try and do the input fields/HTML and output the stored values only for the fields.

Reply:
This doesn't work as it eventually tries to call `$field->displayValue` or `$field->display` which tries to render the HTML values.

I propose having an argument for `$field_group->display` to add `'markup' => 'value_only'` or something similar (maybe specific keys?) that fetches the fields values for that group in particular as an array of sorts that returns `array( 'field_name_1' => 'My awesome field value', 'company' => 'Paid Memberships Pro');` and so on. While you can just get this through get_user_meta if you know the keys directly it may make it easier for cases where we want to fetch field values for a particular group to cross-reference the value.

Reply:
Can also set the `$field->displayValue( $value, false )` inside the call to output just the field and not the HTML part which could work. I do think we'd need a trigger to enable this programmatically as we'd want to use it for the HTML generation or just the values. Maybe the markup value would trigger this logic.